docs/static/examples/FABButton.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import { StyleSheet, SafeAreaView, View, Pressable, Text } from 'react-native';
+import Animated, {
+  withDelay,
+  interpolate,
+  useAnimatedStyle,
+  useSharedValue,
+  withSpring,
+  withTiming,
+} from 'react-native-reanimated';
+
+const AnimatedPressable = Animated.createAnimatedComponent(Pressable);
+
+const SPRING_CONFIG = {
+  duration: 1200,
+  overshootClamping: true,
+  dampingRatio: 0.8,
+};
+
+const OFFSET = 60;
+
+const FABButton = ({ animatedStyles, buttonLetter }) => (
+  <AnimatedPressable style={[animatedStyles, styles.shadow, styles.button]}>
+    <Animated.Text style={styles.content}>{buttonLetter}</Animated.Text>
+  </AnimatedPressable>
+);
+
+export default function App() {
+  const isExpanded = useSharedValue(false);
+
+  const handlePress = () => {
+    isExpanded.value = !isExpanded.value;
+  };
+
+  const generateButtonAnimatedStyle = (index) =>
+    useAnimatedStyle(() => {
+      const moveValue = interpolate(
+        Number(isExpanded.value),
+        [0, 1],
+        //highlight-next-line
+        [0, OFFSET * index]
+      );
+      const translateValue = withSpring(-moveValue, SPRING_CONFIG);
+      //highlight-next-line
+      const delay = index * 100;
+      return {
+        transform: [
+          { translateY: translateValue },
+          {
+            scale: withDelay(
+              delay,
+              //highlight-next-line
+              withTiming(interpolate(moveValue, [OFFSET * index, 0], [1, 0]))
```

Comment:
I'm not sure if 'interpolate' is necessary here because the value of interpolation function will evaluate only one during the animation start 🤔 Correct me if I'm wrong.